src/runtime/objmodel.cpp

```diff
@@ -405,6 +412,54 @@ void BoxedClass::finishInitialization() {
     commonClassSetup(this);
 }
 
+LocalShape BoxedClass::computeLocalShape() {
+    if (local_shape.isUninitialized()) {
+        std::vector<std::pair<llvm::StringRef, int>> attrs;
+
+        // iterate over all class attributes, storing the pairs of name, offset into attrs
+        auto hcls = getHCAttrsPtr()->hcls;
+        llvm::StringMap<int>::const_iterator it;
+        for (it = hcls->getStrAttrOffsets().begin(); it != hcls->getStrAttrOffsets().end(); it++) {
+            attrs.emplace_back(it->first(), it->second);
+        }
+        // sort the vector to hopefully give a deterministic ordering
+        std::sort(attrs.begin(), attrs.end(), [](std::pair<llvm::StringRef, int> a, std::pair<llvm::StringRef, int> b) {
+            return a.second < b.second;
+        });
+
+        // compute a hash of all attribute names
+        SHA256OStream hash_stream;
+        for (auto& s : attrs) {
+            hash_stream << s.first;
```

Comment:
I think this can't distinguish between a class with attributes `["a", "bc"]` and `["ab", "c"]`.  There might be a better way to create the aggregated hash, but it seems like a simple fix for now is do something like `hash_stream << s.first.size() << s.first()`, which would make unambiguous the mapping from stream -> attribute set.